Sources/Build/BuildPlan.swift

```diff
@@ -509,25 +537,6 @@ public class BuildPlan: SPMBuildCore.BuildPlan {
             }
         }
 
-        var productMap: [ResolvedProduct: ProductBuildDescription] = [:]
-        // Create product description for each product we have in the package graph that is eligible.
-        for product in graph.allProducts where product.shouldCreateProductDescription {
-            guard let package = graph.package(for: product) else {
-                throw InternalError("unknown package for \(product)")
-            }
-            // Determine the appropriate tools version to use for the product.
-            // This can affect what flags to pass and other semantics.
-            let toolsVersion = package.manifest.toolsVersion
-            productMap[product] = try ProductBuildDescription(
-                package: package,
-                product: product,
-                toolsVersion: toolsVersion,
-                buildParameters: buildParameters,
-                fileSystem: fileSystem,
-                observabilityScope: observabilityScope
-            )
-        }
-
```

Comment:
This just got moved up so that we can find macro products when constructing the `SwiftTargetBuildDescription` of targets which require any macros to build.